pkg/cmd/server/apis/config/validation/oauth.go

```diff
@@ -354,9 +356,35 @@ func ValidateGitHubIdentityProvider(provider *configapi.GitHubIdentityProvider,
 		}
 	}
 
+	if hostname := provider.Hostname; len(hostname) != 0 {
+		hostnamePath := fieldPath.Child("hostname")
+
+		if hostname == "github.com" || strings.HasSuffix(hostname, ".github.com") {
+			validationResults.AddErrors(field.Invalid(hostnamePath, hostname, "cannot equal [*.]github.com"))
+		}
+
+		if !isValidHostname(hostname) {
```

Comment:
no scheme, no port?

Reply:
GHE does not let you configure that.  Https with no port only.

Reply:
(based on the limited feedback the UI provides - the configuration settings are very limited)